### PR TITLE
Enhancement for Issue 3: Adding ARM build support

### LINKED ISSUE #3: Adding ARM Architecture
So, I've been fighting with this for a few days. I'm not sure if this project is still being supported, but I thought I'd take a shot and try and see if you might know what is going wrong here.

I am trying to add ARM support. The included `buildmips.sh` and `buildmipsel.sh` work great, so I installed the ARM cross compile tools and set up a similar `buildarm.sh` script:

```bash
#!/bin/sh

export ARCH=arm
TARGET=$1

# Sets up toolchain environment variables for various mips toolchain

warn()
{
	echo "$1" >&2
}

if [ ! -z $(which arm-linux-gnueabi-gcc-5) ];
then
	export CC=$(which arm-linux-gnueabi-gcc-5)
else
	warn "Not setting CC: can't locate arm-linux-gnueabi-gcc-5."
fi

if [ ! -z $(which arm-linux-gnueabi-ld) ];
then
	export LD=$(which arm-linux-gnueabi-ld)
else
	warn "Not setting LD: can't locate arm-linux-gnueabi-ld."
fi

if [ ! -z $(which arm-linux-gnueabi-ar) ];
then
	export AR=$(which arm-linux-gnueabi-ar)
else
	warn "Not setting AR: can't locate arm-linux-gnueabi-ar."
fi


if [ ! -z $(which arm-linux-gnueabi-strip) ];
then
	export STRIP=$(which arm-linux-gnueabi-strip)
else
	warn "Not setting STRIP: can't locate arm-linux-gnueabi-strip."
fi

if [ ! -z $(which arm-linux-gnueabi-nm) ];
then
	export NM=$(which arm-linux-gnueabi-nm)
else
	warn "Not setting NM: can't lcoate arm-linux-gnueabi-nm."
fi


make $TARGET || exit $?
```

> Side note: arm-linux-gnueabi-gcc-5 is being used instead of arm-linux-gnueabi-gcc because, using some code from the devtty0's blog to build an nvram hook, I found it was looking for libc.so.6 instead of the device's libc.so.0. I also tried to build against the devices libc.so.0 but was quickly out of my depth

Everything compiles fine, but when I execute:

```
sudo chroot . ./qemu-arm-static -E LD_PRELOAD="/libnvram-faker.so" sbin/version
```

I get the following error:
```
sbin/version: symbol 'initialize_ini': can't resolve symbol

sbin/version: symbol 'end': can't resolve symbol
```

Both of these functions obviously exist in `nvram-faker/nvram-faker.c`, I don't understand :(

### PR DESCRIPTION
**Pull request by decidedlygray**
Issue #3 - Request to merge changes to allow for building for ARM

### CODE REVIEW

File: buildarm.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+# Sets up toolchain environment variables for arm toolchain
+# Tested against armv7-eabihf, glibc from toolchains.free-electrons.com
+
+warn()
+{
+	echo "$1" >&2
+}
+
+if [ ! -z $(which arm-linux-gcc-5.4.0) ];
```
Comment (zcutlip) [inline buildarm.sh]: can we get rid of the hard-coded version? there should be a symlink for a non-versioned name of the binary like `arm-linux-gcc` or something

**Comment by zcutlip** [comment]
sorry, I know this is a few years old. would you like it merged still?

**Comment by decidedlygray** [comment]
Hey I totally didn't see your comments before. I saw them by chance today while checking old issues. Yes I would like to merge still, I need to get rid of the hardcoded version. It's been a while since I've been back to this

**Comment by zcutlip** [comment]
I made the edit to the arm shell script and merged. should be good to go

**Pull request merged**